### DISCUSSION TITLE
Use stable collections

### DISCUSSION THREAD

**Discussion by anastr** [patrykandpatrick/vico#336] [Ideas]
As it's mentioned in the [Compose Stability](https://developer.android.com/jetpack/compose/performance/stability#summary).

> Compose always considers collection classes unstable, such as `List`, `Set` and `Map`. This is because it cannot be guaranteed that they are immutable. You can use [Kotlinx immutable collections](https://developer.android.com/jetpack/compose/performance/stability/fix#immutable-collections) instead or annotate your classes as `@Immutable` or `@Stable`.

Since chart libraries depends heavily on collections and animations, they should use immutable collections to improve performance and reduce unnecessary recompositions.

[Kotlinx Immutable Collections](https://github.com/Kotlin/kotlinx.collections.immutable) is the best library to be used in this situation.

**Comment by patrickmichalik** [comment]
Hello! Thanks for the suggestion. We’ll be looking into this.

**Comment by Gowsky** [comment]
As of Vico 2, this is resolved. Vico now uses strong skipping mode, and the required stability adjustments have been made.
